### ISSUE TITLE
New Operator: GELU

### ISSUE THREAD

**Issue by jonas-doevenspeck** [onnx/onnx#4933]
# New Operator

### Describe the operator

Activation function: Gaussian Error Linear Unit (GELU): https://paperswithcode.com/method/gelu

### Can this operator be constructed using existing onnx operators?

Current export behavior: 

![image](https://user-images.githubusercontent.com/103593150/220924683-5da1ca1b-0864-43fd-9ec3-804d8be8b2a1.png)

### Is this operator used by any model currently? Which one?

It is used in recent popular models such as:
- Stable Diffusion: https://github.com/huggingface/diffusers/blob/f5ccffecf7c9c7b08c39043d1c9528be2c5cfecc/src/diffusers/models/attention.py#L395
- ConvNext: https://github.com/pytorch/vision/blob/ed48bb1c8c5cbb4dfaf18f4c27a47c42ce2bde27/torchvision/models/convnext.py#L55
- BERT, GPT-3, ... 

### Are you willing to contribute it? (Y/N)

N

**Comment by gramalingam** [comment]
What is the question?

Yes, Gelu can be defined in terms of existing ONNX operators. For example, onnxruntime has a custom op, defined as a function in terms of other onnx ops.

Is this a request/suggestion that Gelu be added to ONNX op list? That may be a very reasonable thing to do.

**Comment by jonas-doevenspeck** [comment]
@gramalingam the question is indeed to add GELU to the ONNX op list. 
Thank you for considering the request.

**Comment by JuanuMusic** [comment]
This would be SO useful

**Comment by AlexandreEichenberger** [comment]
I concur, it would be quite helpful, in part also because there exists faster approx of GeLU, see https://pytorch.org/docs/stable/generated/torch.nn.GELU.html for example.

It is used also in Roberta, another important model.

So not only we would be able to cover more models, but having it may also enable us to lower the GeLU to a faster version that the current ERF based implementation.

@gramalingam how do we get the ball rolling on this one?

**Comment by gramalingam** [comment]
It's a matter of creating a PR. It should be straightforward to migrate the op definition that exists in [onnxruntime](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/graph/contrib_ops/contrib_defs.cc#L550). Is there any volunteer? I can also bring it up in the weekly sync meetings or SIG meeting to find some volunteer.

**Comment by pranshupant** [comment]
I would like to volunteer for this as this would be quite a useful addition to the existing operator set. Can I just start working on it and submit a PR or do I need to do some more work before getting started?
Is this guide still up to date? [Proposing and submitting a new operator or function to ONNX](https://github.com/onnx/onnx/blob/main/docs/AddNewOp.md#proposing-and-submitting-a-new-operator-or-function-to-onnx-)

**Comment by gramalingam** [comment]
Summarizing a few points that were brought up in yesterday's SIG meeting:

* It would be useful to define both the regular Gelu, as well as its fast-approximation based on Tanh.
* Two options were discussed: for defining these two as two separate function-ops, or as one op with an attribute to choose between the two. The consensus was to define one op, with an attribute, similar to what pytorch does.
* It should be defined as a function-op: as a context-dependent function-op, that uses the attribute to return one of the two possible function-body definitions.

**Comment by gramalingam** [comment]
> I would like to volunteer for this as this would be quite a useful addition to the existing operator set. Can I just start working on it and submit a PR or do I need to do some more work before getting started? Is this guide still up to date? [Proposing and submitting a new operator or function to ONNX](https://github.com/onnx/onnx/blob/main/docs/AddNewOp.md#proposing-and-submitting-a-new-operator-or-function-to-onnx-)

Hi, yes, that document is a good starting point. If you have any further questions, please do not hesitate to ask us, thanks very much for volunteering to do this.

**Comment by justinchuby** [comment]
Thank you @pranshupant!

**Comment by pranshupant** [comment]
> Thank you @pranshupant!

Working on this was fun! Would love to contribute more in the future.

**Comment by justinchuby** [comment]
FYI there's a list of issues we need help with: https://github.com/onnx/onnx/issues?q=is%3Aopen+is%3Aissue+label%3A%22contributions+welcome%22+sort%3Aupdated-desc. Any help would be appreciated
